_posts/en/newsletters/2019-03-05-newsletter.md

```diff
@@ -0,0 +1,469 @@
+---
+title: 'Bitcoin Optech Newsletter #36'
+permalink: /en/newsletters/2019/03/05/
+name: 2019-03-05-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter links to the announcement of the C-Lightning 0.7
+upgrade, notes a service outage for the Bitcoin-Dev mailing list, and
+describes a proposed soft fork to eliminate several old problems in the
+Bitcoin consensus protocol.  Also included are summaries of notable
+commits in popular Bitcoin infrastructure projects.
+
+## Action items
+
```

Comment:
Do we want to explicitly call out that anyone aware of any Bitcoin user, transaction, or wallet software that has ever used any of the opcodes that are now non-standard and will soon be obsoleted should come forward ASAP? Otherwise the money would be lost forever. Do we know if a lot of communication was done when these opcodes were made non-standard? If yes, then perhaps we could instead, or in addition, link to this communication and any kind of feedback/debate.

Reply:
So the mining change is new and miners will need to upgrade to guarantee compliance with it (although 95% will need to signal for BIP9 support anyway).  As we discussed OOB, the OP_CODESEPARATOR and FindAndDelete being non-standard is relatively new in P2SH, although it's been non-standard in bare scriptPubKey since 2010.  Everything else has been non-standard since 2010 or 2011, which I think is enough lead time.  About OP_CODESEPARATOR, we [previously wrote](https://bitcoinops.org/en/newsletters/2019/01/08/):

> Also discussed is whether the OP_CODESEPARATOR opcode should be disabled in a script update that supports MAST (e.g. via Taproot). That opcode is not in common use, but if you plan to use it in future Script versions, you should comment on the thread.

Also, I don't think it really counts, but our trial-run "[newsletter 0](https://bitcoinops.org/en/newsletters/2018/06/08/)" also mentioned it:

> Check for use of the CodeSeparator opcode: Bitcoin Core 0.16.1, expected to be released within the next week or so, will no longer relay legacy (non-segwit) transactions using this opcode. As far as Bitcoin Core contributors know, nobody uses this opcode, but if your organization does use it or plans to use it, you should immediately contact a protocol developer or optech team member ASAP to inform them of this. Ultimately it is possible that this opcode will be removed from the Bitcoin Protocol for legacy transactions via a soft fork.

I think the fact that we've pretty much dedicated an entire newsletter to this soft fork (and will likely be mentioning it several more times, not to mention all the other people who will mention it), is probably enough notice for anyone who has ever used these *really advanced* features to realize their peril.

Reply:
Agree!